core/string/ustring.cpp

```diff
@@ -4093,6 +4093,43 @@ String String::replacen(const char *p_key, const char *p_with) const {
 	return new_string;
 }
 
+String String::replace(const Vector<String> &p_keys, const String &p_with) const {
+	String new_string;
+
+	const int len = length();
+	for (int src_pos = 0; src_pos < len; src_pos++) {
+		bool replaced = false;
+		const char32_t *src = &operator[](src_pos);
+
+		for (const String &key : p_keys) {
+			const int key_len = key.length();
+			if (src_pos + key_len > len)
```

Comment:
```suggestion
			if (src_pos + key_len > len || key_len == 0)
```

If `p_keys` contains an empty string, `key_len` being zero could cause the `src_pos += key_len - 1;` line below to set `src_pos` back one, cancelling out the loop increment and repeatedly adding `p_with`